packages/ts/hilla-file-router/src/vite-plugin/collectRoutesFromFS.ts

```diff
@@ -0,0 +1,68 @@
+import { opendir } from 'node:fs/promises';
+import { basename, extname, relative } from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+export type RouteMeta = Readonly<{
+  path: string;
+  file?: URL;
+  layout?: URL;
+  children: RouteMeta[];
+}>;
+
+export type CollectRoutesOptions = Readonly<{
+  extensions: readonly string[];
+  parent?: URL;
+}>;
+
+function cleanUp(blank: string) {
+  return blank
+    .replaceAll(/\{\.{3}(.+)\}/gu, '$1')
+    .replaceAll(/\{{2}(.+)\}{2}/gu, '$1')
+    .replaceAll(/\{(.+)\}/gu, '$1');
+}
```

Comment:
I guess a slightly different black magic is in effect here, which can leverage from some comments for the maintainers.